### ISSUE TITLE
Custom or Polymorphic Type Support

### ISSUE THREAD

**Issue by bradygaster** [domaindrivendev/Ahoy#120]
There has long been some debate around the provision of a method to output polymorphic types in the JSON. Is there an ETA for providing it via the Swagger 2.0/OAI specification's provisions? 

Also - is there any guidance on adding custom models that should be shown in the definitions? I've tried creating an IDocumentFilter and using the context.SchemaRegistry.GetOrRegister() method, but the context.SchemaRegistry object is always null. Any guidance here would be greatly appreciated!

**Comment by domaindrivendev** [comment]
Firstly, SchemaRegistry in the document filter should not be null - that's an outright bug and needs to be fixed - https://github.com/domaindrivendev/Ahoy/blob/master/src/Swashbuckle.SwaggerGen/Generator/SwaggerGenerator.cs#L56

Secondly, on the topic of polymorphism, I'm really reluctant to tackle it and it's associated complexity right now because, although it's kind of ... sort of ... but not really the way you might think, supported in the spec, it's not currently supported by the swagger-ui. In fact, if I were to update Swashbuckle to use the "allOf" construct for describing inheritance hierarchies instead of just flattening out inherited properties unto the subclass, then the current UI experience would actually be degraded because those inherited properties would no longer be visible.

So, in summary, it's low on the list of priorities until adequate support is baked into the swagger-ui. For client code-gen scenarios, which I'm guessing you're most interested in here, I would say adding the additional sub-classes via a DocumentFilter makes the most sense (just need to get that null ref fixed).

**Comment by bradygaster** [comment]
I totally understand the reluctance with the variant behavior in Swagger and with it potentially being a moving target. That said, I think this is something a lot developers would like to see happen, and it would mitigate one of the biggest detractors from SB's use. Lack of support for it in the UI or not, i think it would be a great feature to add. I've got a specific case with an app I'm building currently that would make huge use of it if it were in there. For folks who want to do it, do you suggest an alternative? I think a demo project would go a long way to educating people on workarounds or alternatives.

**Comment by bradygaster** [comment]
One more question for @domaindrivendev - should I create an issue to associate with the "nullness?" If it were not null I would've probably been able to implement some sort of workaround, though not-as-awesome as true polymorphism would be.

**Comment by domaindrivendev** [comment]
@bradygaster  - please do create a separate ticket to address the "null" issue

**Comment by domaindrivendev** [comment]
And actually the polymorphic issue is really a dup of #2 so am going to close this one and refer to that for tracking

**Comment by bradygaster** [comment]
@domaindrivendev created a new issue [here](https://github.com/domaindrivendev/Ahoy/issues/132) for this. Thanks!
